src/utils/getProfileData.ts

```diff
@@ -54,12 +54,30 @@ function requestAccessToken({origin}: URL): Promise<SubjectAccessToken> {
 function formatLiquidProfileData(
   entries: ProfileNode[],
 ): FormattedProfileNode[] {
-  return entries.map(function(entry: ProfileNode) {
-    if (!entry.partial.includes(':')) {
-      entry.partial = `snippet:${entry.partial}`;
+  return entries.map((entry: ProfileNode) => {
+    const nameParts = entry.partial.split('/');
+    let name = '';
+    let filename = null;
+    if (nameParts.length === 1 && !entry.partial.includes(':')) {
+      name = `snippet:${entry.partial}`;
+      filename = `snippets/${entry.partial}.liquid`;
+    } else if (/shopify:\/\/apps/.test(entry.partial)) {
+      name = `app-${nameParts[4].slice(0, -1)}:${nameParts[5]}:${nameParts[3]}`;
+      entry.code = entry.code || entry.partial;
```

Comment:
Should `filename` also be set here?

Reply:
No .. because it won't exist in theme files. A null `filename` is how we are determining if we should create an anchor link

Reply:
maybe I should rename that to filepath

Reply:
I see that the type is set to `string | null`, that makes sense then 👍 `filepath` would  make sense in this context.